v4/core/cp4d_authenticator.go

```diff
@@ -172,9 +172,13 @@ func (authenticator *CloudPakForDataAuthenticator) getToken() (string, error) {
 		}
 	}
 
+	authError := &AuthenticationError{}
```

Comment:
I think we should initialize authError only if we're going to return an actual error.
```suggestion
```